airflow/contrib/operators/databricks_operator.py

```diff
@@ -0,0 +1,146 @@
+from airflow.models import BaseOperator
+
+POLL_SLEEP_PERIOD_SECONDS = 5
+LINE_BREAK = ("-" * 80)
+
+
+class DatabricksSubmitRunOperator(BaseOperator):
+    """
+    Submits an ephemeral run to Databricks and waits for it to complete
+    successfully.
+
+    For more information about Databricks ephemeral runs look at
+    https://docs.databricks.com/api/latest/jobs.html#runs-submit
+    """
+    def __init__(self,
+                 spark_jar_task=None,
+                 notebook_task=None,
+                 new_cluster=None,
+                 existing_cluster_id=None,
+                 libraries=[],
+                 run_name=None,
+                 timeout_seconds=0,
+                 extra_api_parameters={},
+                 databricks_conn_id='databricks_default',
+                 **kwargs):
+        """
+        Create a new `DatabricksSubmitRunOperator`. Note that the named
+        parameters to this operator match the parameters exposed by
+        `api/2.0/runs/submit`.
+
+        As a result, one way to instantiate a `DatabricksSubmitRunOperator`
+        is to pass the same JSON object used to call `api/2.0/runs/submit`
+        into the `DatabricksSubmitRunOperator`. For example:
+
+        ```
+        params = {
+          "new_cluster": {
+            "spark_version": "2.0.x-scala2.10",
+            "node_type_id": "r3.xlarge",
+            "aws_attributes": {
+              "availability": "ON_DEMAND"
+            },
+            "num_workers": 2
+          },
+          "libraries": [
+            {
+              "jar": "dbfs:/test.jar"
+            },
+            {
+              "maven": {
+                "coordinates": "org.jsoup:jsoup:1.7.2"
+              }
+            }
+          ],
+          "spark_jar_task": {
+            "main_class_name": "com.databricks.ComputeModels"
+          }
+        }
+        DatabricksSubmitRunOperator(task_id='run-1', **params)
+        ```
+        """
+        super(DatabricksSubmitRunOperator, self).__init__(**kwargs)
+        self.spark_jar_task = spark_jar_task
+        self.notebook_task = notebook_task
+        self.new_cluster = new_cluster
+        self.existing_cluster_id = existing_cluster_id
+        self.libraries = libraries
+        self.run_name = kwargs['task_id'] if run_name is None else run_name
+        self.timeout_seconds = timeout_seconds
+        self.extra_api_parameters = extra_api_parameters
+        self.databricks_conn_id = databricks_conn_id
+        self._validate_parameters()
+
+    def _validate_parameters(self):
+        self._validate_oneof(self.spark_jar_task, self.notebook_task)
+        self._validate_oneof(self.new_cluster, self.existing_cluster_id)
+
+    def _validate_oneof(self, param_a, param_b):
+        if param_a is not None and param_b is None:
+            pass
+        elif param_a is None and param_b is not None:
+            pass
+        else:
+            raise AirflowException
```

Comment:
we need to provide better error message - I think this would be a common problem people would encounter.